lib/backend/compat/compat.go

```diff
@@ -251,6 +313,44 @@ func (c *ModelAdaptor) listNodes(l model.NodeListOptions) ([]*model.KVPair, erro
 	return results, nil
 }
 
+// listBlock returns list of KVPairs for Block, includes making sure
+// backwards compatiblity. See getBlock for more details.
+func (c *ModelAdaptor) listBlock(l model.BlockListOptions) ([]*model.KVPair, error) {
+
+	// Get a list of block KVPairs.
+	bal, err := c.client.List(l)
+	if err != nil {
+		return nil, err
+	}
+
+	// Go through the list to make sure Affinity field has a proper value,
+	// and map the value to Affinity if deprecated HostAffinity field is used
+	// by calling ensureBlockAffinity, and update the KVPair slice to return.
+	for idx, bakv := range bal {
+		bal[idx] = ensureBlockAffinity(bakv)
```

Comment:
I think this is OK, but be really careful about modifying arrays while you're iterating over them - especially if the array size changes.

I'd feel more comfortable if you generated a new array here rather than re-using the `bal` array.

Also, what does `bal` and `bakv` stand for?  Slightly more descriptive names would be nice :)